src/thumbnailer.cc

```diff
@@ -60,9 +59,15 @@ Thumbnailer::Status Thumbnailer::GenerateAnimationNoBudget(
 
   // Fill the animation.
   for (auto& frame : frames_) {
-    if (!WebPAnimEncoderAdd(enc_, &frame.pic, frame.timestamp_ms, &config_)) {
+    WebPPicture new_pic;
```

Comment:
I do not see why a copy is done here neither at line 225.

Reply:
Since I want to save the original frames for calling `WebPPictureDistortion` later. I think if I used `WebPAnimEncoderAdd(enc_, &frame.pic, frame.timestamp_ms, &frame.config)` the contents of `frame.pic` would be changed.

Reply:
Can you add a comment to explain this?

Reply:
From the doc of `WebPAnimEncoderAdd()`:
`frame - (in/out) frame data in ARGB or YUV(A) format. If it is in YUV(A) format, it will be converted to ARGB, which incurs a small loss.`
So if `frame.pic.use_argb` is true, the pixels should not be modified and the copy unnecessary. However it might be safer to copy it anyway and keep the optimization for later; then a comment as Maryla suggested would be appreciated. I agree that the libwebp API appears confusing and inconsistent, it is hard to know what is modified and when.